crates/services/p2p/src/decay.rs

```diff
@@ -0,0 +1,101 @@
+use libp2p::{
+    core::Endpoint,
+    swarm::{
+        derive_prelude::FromSwarm,
+        dummy,
+        ConnectionDenied,
+        ConnectionId,
+        NetworkBehaviour,
+        THandler,
+        THandlerInEvent,
+        THandlerOutEvent,
+        ToSwarm,
+    },
+    Multiaddr,
+    PeerId,
+};
+use std::{
+    task::{
+        Context,
+        Poll,
+    },
+    time::Duration,
+};
+use tokio::time::{
+    self,
+    Interval,
+};
+
+const REPUTATION_DECAY_INTERVAL_IN_SECONDS: u64 = 1;
+
+#[derive(Debug, Clone)]
+pub enum DecayEvent {
+    /// Informs p2p service / PeerManager to perform reputation decay of connected nodes
+    PerformDecay,
+}
+
+pub struct Behavior {
+    decay_interval: Interval,
```

Comment:
Why can't we just put `decay_interval` on to `FuelBehaviour`?

Reply:
in order for the #[derive(NetworkBehaviour] macro to work all fields of the structs must impl Network behaviour

Reply:
I see. I'm not really sure what the purpose of this issue was then tbh. I don't feel like moving each part to separate behaviors is buying us much.

Reply:
makes sense I can close the issue

Reply:
I'd just leave it for now. Maybe someone else remembers the context. It's possible to implement `NetworkBehavior` manually, so maybe that's the play? but it's low priority anyway.